Feature medical non medical employees

lomamech opened the pull request:
feature(Distinguish between Medical and Non Medical)

- Update Schema add property is_medical in Employee table
- Add Property is_medical in Form for employee registration
- Add Property is_medical in Employee Registry
- Update Integration and E2E test, add property is_employee
closes #1630


refactor(Employee Registry)
- Remove Component bhRenderDropDown
- Add grid Save State Option
- Add column Configuration
- Fix error on build Employee Report Use sex instead sexe
- Set correct column in Employee Handlebare
closes #1745

- Implement component bhGradeSelect
- Implement component bhFonctionSelect
- Use component bhServiceSelect in Search Employee Registry
- Implement E2E test to check new Component

lomamech committed:
feature(Distinguish between Medical and Non Medical)
- Update Schema add property is_medical in Employee table
- Add Property is_medical in Form for employee registration
- Add Property is_medical in Employee Registry
- Update Integration and E2E test, add property is_employee

closes #1630

lomamech committed:
refactor(Employee Registry)
- Remove Component bhRenderDropDown
- Add grid Save State Option
- Add column Configuration
- Fix error on build Employee Report Use sex instead sexe
- Set correct column in Employee Handlebare

closes #1745

jniles commented on client/src/js/components/bhRendererDropdown.js:
```diff
@@ -1,64 +0,0 @@
-angular.module('bhima.components')
```
:dancer:

jniles commented on client/src/modules/employees/employee.service.js:
```diff
@@ -89,5 +91,16 @@ function EmployeeService(Filters, $uibModal, Api, AppCache) {
     }).result;
   }
 
+  function download(type) {
```
Nice :+1:

jniles commented on client/src/modules/employees/registry/registry.js:
```diff
@@ -152,7 +164,7 @@ function EmployeeRegistryController(Employees, Notify,
     toggleLoadingIndicator();
 
-    // hook the returned patients up to the grid.
+    // hook the returned employeess up to the grid.
```
:dancer:

jniles commented on client/src/modules/employees/registry/registry.html:
```diff
@@ -7,41 +7,62 @@
 </div>
 
-<div class="flex-util" style="min-height : 35px; padding-top : 7px; max-height: initial">
+<div class="flex-util bh-filter-bar">
```
:+1:

jniles commented on client/src/modules/employees/registry/registry.js:
```diff
@@ -2,35 +2,33 @@ angular.module('bhima.controllers')
-];
+  '$state', 'EmployeeService', 'NotifyService', 'AppCache',
+  'util', 'ReceiptModal', 'uiGridConstants', '$translate',
+  'GridColumnService', 'GridSortingService', 'bhConstants',
```
Do you need `GridSorting`?  It doesn't look like you use it.

jniles commented on client/src/modules/employees/registry/registry.js:
```diff
@@ -2,35 +2,33 @@ angular.module('bhima.controllers')
-  'GridColumnService', 'bhConstants', 'FilterService',
-];
+  '$state', 'EmployeeService', 'NotifyService', 'AppCache',
+  'util', 'ReceiptModal', 'uiGridConstants', '$translate',
```
Do you need `$translate`?

jniles commented on client/src/modules/employees/registry/search.modal.html:
```diff
@@ -66,58 +66,46 @@
-            </select>
+          <div class="radio">
+            <label>
+              <input type="radio" name="sex" id="female" value="0" ng-model="ModalCtrl.searchQueries.is_medical">
```
Why is this name "sex" and id="female"?

lomamech replied:
I will delete the id property but this code is old

jniles replied:
:+1: Thanks!

jniles commented on client/src/modules/employees/registry/search.modal.html:
```diff
@@ -66,58 +66,46 @@
-            </label>
-
+          <bh-grade-select
+            grade-uuid="ModalCtrl.searchQueries.grade_id"
```
Why is this `grade-uuid`, but it takes in a `grade_id`?  Is the grade a uuid or an id?

lomamech replied:
Because in API in server part all sql request Make the Jointure like this
````
JOIN grade ON employee.grade_id = grade.uuid
````
it's for that i have send grade_id instead grade_uuid

jniles replied:
O.o.  Okay, we should probably fix this.  It looks like it is stored in the database as a `uuid`, but the column is called `id` for some reason.

jniles commented on client/src/modules/employees/registry/search.modal.html:
```diff
@@ -66,58 +66,46 @@
-
+          <bh-grade-select
+            grade-uuid="ModalCtrl.searchQueries.grade_id"
+            name="grade_uuid"
```
I don't think you require a `name` on the `bhGradeSelect` component.

jniles commented on client/src/modules/employees/registry/search.modal.html:
```diff
@@ -66,58 +66,46 @@
-            </label>
+          <bh-fonction-select
+            fonction-id="ModalCtrl.searchQueries.fonction_id"
+            name="fonction_id"
```
I don't think you have defined a `name` on the `bhFonctionSelect` component.

jniles commented on server/controllers/payroll/reports/registrations.handlebars:
```diff
@@ -38,9 +38,9 @@
             <td>{{this.code}}</td>
             <td>{{this.display_name}}</td>
-            <td>{{this.sexe}}</td>
+            <td>{{this.sex}}</td>
```
:dancer:

jniles commented on test/end-to-end/employees/registry.spec.js:
```diff
@@ -66,4 +66,17 @@ describe('Employees Registry', () => {
     employeeRegistryPage.employeeCount(employeeCount, `The number of filtered employee should be ${employeeCount}`);
   });
+
+  it('should find Search for employees belonging to Service Administration, to funtion Infirmier and Grade 1.1', () => {
```
:+1:

jniles reviewed (commented):
Really good work!

There are a few clean up changes to make, nothing major.  After this, we can start to implement some of the Ratio Analysis requested previously.

lomamech committed:
refactor(Employees Registry)
- Remove unused controller injection
- Sanitaze component form
- Remove not necessary property
- Remove property name in some component

lomamech committed:
fix(Employee Registry)
- Set property id in select sex, used in e2e test
- Fix and remove bad id in select employee medical staff`

lomamech committed:
Fix conflict with branch Master

jniles reviewed (approved):
LGTM!

bors r+
